### ISSUE TITLE
[Feature] Add deleted username to reserved username

### ISSUE THREAD

**Issue by 6543** [go-gitea/gitea#13867]
If a user get deleted, add its login name into reserved user list

depend on #6280

**Comment by lunny** [comment]
Why should we do that? A deleted user should have been disappeared  from the instance entirely.

**Comment by 6543** [comment]
somebody might fine it usefull, it should be an opt-in feature...

ref: https://codeberg.org/Codeberg/Community/issues/356

**Comment by jolheiser** [comment]
To be fair, the solution doesn't necessarily solve the reference issue.

> Anyone can impersonate other people because deleted user's username is reusable

> good1 authored /exampler/helpme/commit/xxxxxxxxxxxxxxxxxx

This is a byproduct of using Git and is irrelevant to a user account. 

For example, see [this commit](https://github.com/jolheiser/Test/commit/027e0bdda19d3fe6e526194251fe94d51479c92a), which was definitely made by tech and not myself impersonating him. :wink:

**Comment by 6543** [comment]
sign your commits if you can ...

**Comment by troyengel** [comment]
Many other services reserve the namespace (post-delete) in order to prevent future "impersonation attacks" (my phrase) in the web space (not the git commit space as outlined above). I joined Codeberg recently and was surprised to find Gitea let me change my username 3 times (or more, I was messing around) and I was able to "reclaim" the same username more than once.

It is common with Github and Gitlab for example to inject an internal redirect from the old namespace to the new namespace as well, which allows for upstream to move a repo (example: repo move from a personal space to an Org space) and all the git forks and git checkouts are invisibly redirected from the old namespace to the new one at the server level.
